compiler/testData/codegen/boxInline/nonLocalReturns/fromArrayGeneratorWithFinally.kt

```diff
@@ -0,0 +1,19 @@
+// FILE: 1.kt
+inline fun runReturning(f: () -> Nothing): Nothing = f()
+
+// FILE: 2.kt
+fun box(): String {
+    var r = ""
+    val x = try {
+        Array<String>(1) {
```

Comment:
1. ...test with additional external layer of Array with try/finallies:
 a. where return jumps to external one 
 b. and to internal one

2. And some test with catch/exception throwing

Reply:
Sorry, I don't understand what you mean...

Reply:
for 1 I meant smth like this
```
// FILE: 2.kt
fun box(): String {
    var r = ""
    val x = try {
        Array<String>(1) ext@{
            try {
                Array<String>(1) {
                    try {
                        runReturning { return@ext/Array "OK" }
                    } finally {
                        r += "O"
                    }
                }[0]
            } finally {
                r += ...
            }

        } )
    } finally {
        r += "K"
    }
    return r
}
```

Reply:
And for second one  original test with catch clause that is executed

Reply:
Done, I guess?

Reply:
Almost, last two suggestion: It's good to have test similar to `fromArrayGeneratorWithFinallyX2.kt`  with return@Array  to inner lambda  (that I meant by second `Array` part in `return@ext/Array`). 

Catch test is also good. Could you add it's copy with catch inside  lambda?

Reply:
I don't really get what's the point in adding tests with `catch` clauses, since no non-local returns could be executed in those.